server/api/room.ts

```diff
@@ -237,6 +238,16 @@ const patchRoom: RequestHandler = async (req, res) => {
 		delete req.body.permissions;
 	}
 
+	if (req.body.title && req.body.title.length > 255) {
+		res.status(401).json({
+			success: false,
+			error: {
+				field: "title",
+				message: "Room title must be 255 characters or less"
+			}
+		});
```

Comment:
This is the problem, you're responding to the request without returning, so execution continues and results in OTT trying to respond to the request twice, and you'll get a `ERR_HTTP_HEADERS_SENT` error and it'll shut down. 

You aren't meant to directly respond with an error like this because of how error prone it is. You should be throwing a `BadApiArgumentException`, like the other validators above. The error handler will automatically respond with the correct json and HTTP status.

Reply:
That's what I figured as you did similar validation on the name in createRoom. I will make those changes.